packages/ckeditor5-upload/src/filerepository.ts

```diff
@@ -243,39 +252,53 @@ export default class FileRepository extends Plugin {
 	 *
 	 * @private
 	 */
-	_updatePendingAction() {
+	private _updatePendingAction(): void {
 		const pendingActions = this.editor.plugins.get( PendingActions );
 
 		if ( this.loaders.length ) {
 			if ( !this._pendingAction ) {
 				const t = this.editor.t;
-				const getMessage = value => `${ t( 'Upload in progress' ) } ${ parseInt( value ) }%.`;
+				const getMessage = ( value: number ) => `${ t( 'Upload in progress' ) } ${ parseInt( value as any ) }%.`;
 
 				this._pendingAction = pendingActions.add( getMessage( this.uploadedPercent ) );
 				this._pendingAction.bind( 'message' ).to( this, 'uploadedPercent', getMessage );
 			}
 		} else {
-			pendingActions.remove( this._pendingAction );
+			pendingActions.remove( this._pendingAction! );
 			this._pendingAction = null;
 		}
 	}
 }
 
-mix( FileRepository, ObservableMixin );
-
 /**
  * File loader class.
  *
  * It is used to control the process of reading the file and uploading it using the specified upload adapter.
  */
-class FileLoader {
+export class FileLoader extends Observable {
+	public readonly id: string;
+
+	private _filePromiseWrapper?: FilePromiseWrapper;
```

Comment:
Is is not undefined through the lifetime of the object. It is reset in `_destroy()` only to help garbage collection (I suppose). I'd make it non-optional and have `( undefined as any )` in `_destroy()`

Reply:
On the other hand, it's private, so it doesn't really matter.